contracts/MetadataViews.cdc

```diff
@@ -415,4 +415,67 @@ pub contract MetadataViews {
             self.socials = socials
         }
     }
+
+    // A view to represent a single field of metadata on an NFT.
+    //
+    // This is used to get traits of individual key/value pairs along with some contextualized data about the trait
+    pub struct Trait {
+        // The name of the trait. Like Background, Eyes, Hair, etc.
+        pub let traitType: String
+
+        // The underlying value of the trait, the rest of the fields of a trait provide context to the value.
+        pub let value: AnyStruct
+
+        // displayType is used to show some context about what this traitType and value represent
+        // for instance, you could set value to a unix timestamp, and specify displayType as "Date" to tell
+        // platforms to consume this trait as a date and not a number
+        pub let displayType: String?
+
+        // Rarity can also be used directly on an attribute.
+        //
+        // This is optional because not all attributes need to contribute to the NFT's rarity.
+        pub let rarity: Rarity?
+
+        init(traitType: String, value: AnyStruct, rarity: Rarity?, displayType: String?) {
+            self.displayType = displayType
+            self.traitType = traitType
+            self.value = value
+            self.rarity = rarity
+        }
+    }
+
+    // helper function to easily convert a dictionary to traits. For NFT collections that do not need either of the
+    // optional values of a Trait, this method should suffice to give them an array of valid traits
+    pub fun dictToTraits(dict: {String: AnyStruct}): [Trait] {
```

Comment:
Should we provide a `{String: String}` dict to traits helper function also? A lot of projects use that

Reply:
Hmm would a `{String: String}` not be able to be converted to `{String: AnyStruct}`? Not opposed to more helper functions either way!

Reply:
You'd probably just have to make all the values strings, even if they represent other types. I think it would still be useful